internal/inventory/app/app.go

```diff
@@ -134,6 +144,64 @@ func (app *InventoryApp) Initialize() error {
 	return app.addRoutes()
 }
 
+// bootstrapAliasConfig loads the aliases from the user's configuration toml and pushes them
+// to the config provider if and only if the Aliases key is not present, or the overwrite
+// config flag is passed via the command line
+// todo: switch to using SDK's custom config capability when upgrade to Ireland
+func (app *InventoryApp) bootstrapAliasConfig(sdkFlags flags.Common) error {
+	overwrite := sdkFlags.OverwriteConfig()
+	app.lc.Debug(fmt.Sprintf("Bootstrapping %s config. OverwriteConfig: %v", aliasesConfigKey, overwrite))
+	// skip checking the existing status if overwrite is enabled
+	if !overwrite {
+		// Note: We need to use GetConfiguration and manually check for the existence of
+		// Aliases because ConfigurationValueExists and GetConfigurationValue both internally
+		// use a method that only returns values for single keys. They return nil for an item that
+		// has a collection of child keys.
+		res, err := app.configClient.GetConfiguration(&inventory.ConsulConfig{})
```

Comment:
Using `HasConfiguration()` will not work either since it is looking at the root path which will already exist from the SDK based config, correct?

Reply:
Correct. That might work if we were to set the base path directly as the `rfid-llrp-inventory/Aliases`, however that would require setting up a second `ConfigurationClient`. I had actually played around with the concept of the base path being `rfid-llrp-inventory/Aliases`, and then watching the path `..`. That actually worked from a Consul perspective, however the EdgeX code that wraps it was unable to convert the keys into go structs. (Even if it did work I would still consider it a hack/workaround anyways).

I can add a comment about `HasConfiguration` as well. And maybe it is worthwhile to submit an Issue against the SDK code? I know you have mentioned Ireland release has some public methods not available in Hanoi that might render this moot anyways.

Reply:
Yea, please add `HasConfiguration` to your comment as that is the one we would expect to be used. THX!